test/Microsoft.AspNetCore.Antiforgery.Test/Internal/DefaultClaimUidExtractorTest.cs

```diff
@@ -97,18 +97,133 @@ public void DefaultUniqueClaimTypes_NotPresent_SerializesAllClaimTypes()
+        public void GetUniqueIdentifierParameters_UsesUpnIfPresent()
+        {
+            // Arrange
+            var identity = new ClaimsIdentity("someAuthentication");
+            identity.AddClaim(new Claim("fooClaim", "fooClaimValue"));
+            identity.AddClaim(new Claim(ClaimTypes.Upn, "upnClaimValue"));
+
+            // Act
+            var uniqueIdentifierParameters = DefaultClaimUidExtractor.GetUniqueIdentifierParameters(new ClaimsIdentity[] { identity });
+
+            // Assert
+            Assert.Equal(new string[]
+            {
+                ClaimTypes.Upn,
+                "upnClaimValue",
+                "LOCAL AUTHORITY",
+            }, uniqueIdentifierParameters);
+        }
+
+        [Fact]
+        public void GetUniqueIdentifierParameters_MultipleIdentities_UsesOnlyAuthenticatedIdentities()
+        {
+            // Arrange
+            var identity1 = new ClaimsIdentity(); // no authentication
+            identity1.AddClaim(new Claim("sub", "subClaimValue"));
+            var identity2 = new ClaimsIdentity("someAuthentication");
+            identity2.AddClaim(new Claim(ClaimTypes.NameIdentifier, "nameIdentifierValue"));
+
+            // Act
+            var uniqueIdentifierParameters = DefaultClaimUidExtractor.GetUniqueIdentifierParameters(new ClaimsIdentity[] { identity1, identity2 });
 
             // Assert
             Assert.Equal(new string[]
             {
                 ClaimTypes.NameIdentifier,
                 "nameIdentifierValue",
+                "LOCAL AUTHORITY",
+            }, uniqueIdentifierParameters);
+        }
+
+        [Fact]
+        public void GetUniqueIdentifierParameters_NoKnownClaimTypesFound_SortsAndReturnsAllClaimsFromAuthenticatedIdentities()
+        {
+            // Arrange
+            var identity1 = new ClaimsIdentity(); // no authentication
+            identity1.AddClaim(new Claim("sub", "subClaimValue"));
+            var identity2 = new ClaimsIdentity("someAuthentication");
+            identity2.AddClaim(new Claim(ClaimTypes.Email, "email@domain.com"));
+            var identity3 = new ClaimsIdentity("someAuthentication");
+            identity3.AddClaim(new Claim(ClaimTypes.Country, "countryValue"));
+            var identity4 = new ClaimsIdentity("someAuthentication");
+            identity4.AddClaim(new Claim(ClaimTypes.Name, "claimName"));
+
+            // Act
+            var uniqueIdentifierParameters = DefaultClaimUidExtractor.GetUniqueIdentifierParameters(
+                new ClaimsIdentity[] { identity1, identity2, identity3, identity4 });
+
+            // Assert
+            Assert.Equal(new List<string>
+            {
+                ClaimTypes.Country,
+                "countryValue",
+                "LOCAL AUTHORITY",
+                ClaimTypes.Email,
+                "email@domain.com",
+                "LOCAL AUTHORITY",
+                ClaimTypes.Name,
+                "claimName",
+                "LOCAL AUTHORITY",
```

Comment:
This test demonstrates the behaviour I complained about [here](https://github.com/aspnet/Antiforgery/pull/52/files#r52409185). Should never mix claims from multiple identities in the same UID.